pkg/nodeconfig/nodeconfig.go

```diff
@@ -258,8 +258,8 @@ func (nc *nodeConfig) SafeReboot(ctx context.Context, node *core.Node) error {
 	if err := nc.Windows.RebootAndReinitialize(); err != nil {
 		return err
 	}
-	// wait for WICD to remove Reboot annotation
-	if err := metadata.WaitForRebootAnnotationRemoval(ctx, nc.client, nc.node.Name); err != nil {
+	// Remove the reboot annotation after we know the reboot occurred successfully
+	if err := metadata.RemoveRebootAnnotation(ctx, nc.client, *nc.node); err != nil {
 		return err
 	}
 
```

Comment:
I don't think this has anything to do with this PR, but are we uncordoning the node too soon before WICD has finished configuring services? The context for the question is for nodes that have already been configured before and has the correct version annotation.

Reply:
I'm not sure. If so, then we'd have to introduce some sort of wait but I am not sure how WICD would communicate that it has finished configuring services

Reply:
Maybe this is fine. If the version annotation matches, we don't kick off a reconfiguration. And if the version annotation has changed, WCMO will wait for WICD to apply the version annotation. Correct?

Reply:
Correct

Reply:
@sebsoto please review and ack if you agree

Reply:
The only reasonable solution I have for that issue is WICD applying a `NeedsCordon`/`NeedsDrain` annotation, and WMCO reacts to its presence or lack of presence. This along the lines of what the MCO does.